SECURITY.md

```diff
@@ -3,13 +3,10 @@
 ## Supported Versions
 
 The list of supported versions can be found
-[here](https://goteleport.com/teleport/download/).
+[here](https://goteleport.com/download/).
```

Comment:
I'd link https://goteleport.com/docs/faq/#which-version-of-teleport-is-supported over the download page.